### DISCUSSION TITLE
Database performance issues

### DISCUSSION THREAD

**Discussion by samuelpisa** [jobrunr/jobrunr#589] [General]
We’ve been using JobRunr scheduled jobs for a while with a MySQL 5.7 database with the following specs:
```
GCP Cloud SQL
Database RAM: 4GB
vCPU: 1

JobRunr JobServer Master: 3
JobRunr Thread Workers: 3
JobRunr Pooling Interval: 15s
```
We have noticed that the database CPU usage have increased exponentially throughout the time after we started using JobRunr. 

We have also noticed that the following query `select jobAsJson from jobrunr_jobs where state = 'SCHEDULED' and scheduledAt <= '****' ORDER BY updatedAt ASC LIMIT 1000 OFFSET 0` is taking too long to complete (for example, at this very moment, we have that query being executed for more than 1 hour).

If we run `select * from jobrunr_jobs_stats` this is the result:
```
total: 3,704,826
scheduled: 3,122,527
enqueued: 0
processing: 0
failed: 7,795
succeeded: 574,504
nbrOfbackgroundJobServers: 5
```
We’ve tried to create a composed index using the state and scheduledAt fields of the jobrunr_jobs table, but again, without success. It seems that JobRunr is having a hard time trying to enqueue the scheduled jobs.

We’re heavily investing on using JobRunr because we hope it would solve many of our recurring problems, but we’re finding it difficult to understand what are the bottlenecks.

Do you have any ideia to help us with our scenario? Thanks in advance.

Kind regards

**Comment by patkujawa-wf** [comment]
What version of jobrunr? There was a recent change to add an index to that table, if I recall correctly.

**Reply by patkujawa-wf** [reply]
Actually, https://github.com/jobrunr/jobrunr/issues/561 was raised by you, @samuelpisa 🙂.

**Comment by rdehuyss** [comment]
Hi @samuelpisa , yes - I do have some assumptions. Would it make sense to setup a short meeting to discuss these? It will be faster than going back and forth here. I'll send you an email with some options to book a time slot.

**Reply by samuelpisa** [reply]
Hi @rdehuyss 
We've upscaled the DB instance and has solved this issue for now.
Still, I've scheduled a meeting to better understand these performance issues.

**Comment by rdehuyss** [comment]
After a call with @samuelpisa , it is clear that they have so many jobs that the index of MySQL could not fit into the 4GB of memory anymore. Increasing the DB instance size solved this.
